.github/workflows/semver_testing.yml

```diff
@@ -0,0 +1,27 @@
+### Ensure that our local testing always passes
+name: Run semver tests
+
+on:
+  pull_request: {}
+
+permissions:
+  contents: read
+  actions: read
+
+jobs:
+  semgrep-tests:
+    name: Run semgrep tests
+    runs-on: ubuntu-latest
+
+    container:
+      image: semgrep/semgrep
+
+    # Skip any PR created by dependabot to avoid permission issues:
+    if: (github.actor != 'dependabot[bot]')
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4    
```

Comment:
Please pin to a specific hash using a tool like [ratchet](https://github.com/sethvargo/ratchet)